src/model/Criteria/CriteriaStorage.php

```diff
@@ -56,4 +58,22 @@ public function getDefaultTableFields(string $table): array
         }
         return $this->defaultFields[$table];
     }
+
+    public function setPrimaryField(string $table, string $field): void
+    {
+        $this->primiaryFields[$table] = $field;
+    }
+
+    public function getPrimaryField(string $table): string
+    {
+        if (!empty($this->primiaryFields[$table])) {
+            return $this->primiaryFields[$table];
+        }
+
+        if (!empty($this->defaultFields[$table])) {
```

Comment:
@mikoczy I'd rather not read default fields while getting the primary field. Default fields were intended as something, that's always selected in the criteria, but it doesn't imply that the first parameter is also the primary.

Can we remove this?

_(no need to make a change here, we'll do it in our internal mono-repo-based PR)_